lib/ClimaCoreTempestRemap/src/onlineremap.jl

```diff
@@ -68,6 +70,83 @@ function remap!(
     return target
 end
 
+# This version of this function is used for distributed remapping
+function remap!(
+    target::Fields.Field,
+    R::LinearMap,
+    source::Fields.Field,
+    context::ClimaComms.AbstractCommsContext,
+    target_mesh::Meshes.AbstractMesh{2},
+    space_o,
+)
+    pid = ClimaComms.mypid(context)
+    nprocs = ClimaComms.nprocs(context)
+
+    target_array = parent(target)
+    source_array = parent(source)
+
+    # Create mapping from global to local xelement indices (for distributed runs)
+    target_elemorder = Meshes.elements(target_mesh)
+    target_elempid = nothing
+    # To make IO easier, we enforce the partitioning to be sequential in the element ordering.
+    target_elempid, target_ranges =
+        Topologies.simple_partition(length(target_elemorder), nprocs)
+    @assert issorted(target_elempid)
+
+    target_local_elem_gidx = target_ranges[pid] # gidx = local_elem_gidx[lidx]
+    target_global_elem_lidx = Dict{Int, Int}() # inverse of local_elem_gidx: lidx = global_elem_lidx[gidx]
+    for (lidx, gidx) in enumerate(target_local_elem_gidx)
+        target_global_elem_lidx[gidx] = lidx
+    end
```

Comment:
we should minimize the amount of work we do in `remap!`: either store `target_global_elem_lidx` in the `LinearMap` case, or expose it as part of `Topology2D`